### DISCUSSION TITLE
How to change the FlowTempDesired?

### DISCUSSION THREAD

**Discussion by pacman789** [john30/ebusd#1135] [Q&A]
Hi everyone,

I bought ebusd V5 adapter and I have saunier duval thema condens boiler 
Out of box it’s working. (wifi mode with docker container)
```
ebusctl info
version: ebusd 23.3.23.3
update check: OK, device firmware 1[4106] available
device: IP:9999, TCP, enhanced, firmware 1.1[3c17].1[3c17]
signal: acquired
symbol rate: 23
max symbol rate: 75
min arbitration micros: 0
max arbitration micros: 207
min symbol latency: 6
max symbol latency: 109
scan: finished
reconnects: 0
masters: 2
messages: 224
conditional: 3
poll: 65
update: 10
address 03: master #11
address 08: slave #11, scanned "MF=Vaillant;ID=BAI00;SW=0701;HW=3302", loaded "vaillant/bai.308523.inc", "vaillant/08.bai.csv"
address 31: master #8, ebusd
address 36: slave #8, ebusd
```
FlowTemp, ReturnTemp, FlowTempDesired, ModulationTempDesired etc. are read.
I have an on/off thermostat (Shelly 1).
My goal is to change the value of FlowTempDesired. I don’t know how to do that. I found out that I have to use SetModeOverride, but I couldn’t figure out how to achieve it. Could you guide me?
```
ebusctl find -w
broadcast id = no data stored
broadcast queryexistence =  (empty for 31fe07fe00 / )
broadcast vdatetime = no data stored
```
I tried this command based on this conversation: (same boiler brand, and same csv/inc file)
https://github.com/john30/ebusd/discussions/781
ebusctl hex 08b5100900007776ffff000100
But I got an error message:
```
ebusctl hex 08b5100900007776ffff000100
ERR: command not enabled
```

My second question is:
Ebusd extract a lot of values that I don't need or ebusd can't read. If I want to delete them, should I just delete the corresponding lines from the bai.308523.inc file?

Thanks for your help.

**Comment by JonesGW-MGD** [comment]
I'm not sure this is going to answer your question, and I'm very new to ebusd and also to heating controls in general but trying to get up to speed as quick as possible and maybe some of this might be useful...!

I'm only using ebusd to read my boiler (Vaillant Ecotec Exclusive combination boiler in my case) so have no experience of writing values.  What cursory exploration I've done, suggests as a minimum you need to check the config csv for that value is writable per the wiki:

https://github.com/john30/ebusd/wiki/4.1.-Message-definition

But - some codes won't be writable - are you sure that FlowTempDesired isn't one of those? In my Vaillant manual (so nothing to do with ebusd), it shows that diagnostic code 5 - Heating target flow temperature (which is mapped as FlowTempDesired in bai.308523.inc) isn't changeable - so you can't alter it from the installer menu on the actual boiler panel for example.

Do you have weather compensation?  (I.e. outside temp sensor)

How I think this works on my boiler (but I can't be certain) is this is the FlowTempDesired is the output of the heat curve algorithm.  So:  the boiler sets an ideal flow temp based on the intersect of the outside temp and the heat curve - it would therefore make sense you can't amend this - but you can amend your heat curve.

Assuming I'm right about weather comp, I wouldn't know what would drive this value if you don't have weather comp.  Maybe if you set d.71 Max. heating target flow temp. But I'm not sure this would mean this was the flow temp, only it was the max the flow temp could be?

FWIW, this is my flow (d.40), return (d.41) and d.05 plot:

<img width="2269" alt="image" src="https://github.com/john30/ebusd/assets/68991992/769747cc-fbed-47de-b3bc-04715d3c6768">

So I guess supplementary question - why are you trying to change your flow temp - are you trying to simulate weather comp?

On your final question - assuming you have cloned the config files locally, yes, you can rem out what you don't need.  (Sometimes when ebusd can't read values - i.e. 'Unknown' it because the code in the config isn't right for your boiler - I've had success in reviewing all the bai files and trying different values to get a mostly complete solution!)

Hope this helps in some small way!

**Comment by apbarratt** [comment]
Thus far, I have not been able to write a single value.  Anything I set (including the desired room temperature) is inevitably ignored and changed back to whatever the thermostat is set to.  It's all been very frustrating to be honest and I'm starting to think my only option is to spend a fortune on buying the official Vaillant smart tech :(

**Reply by JonesGW-MGD** [reply]
without hi-jacking the OP's thread - are you saying you can write values, they just get overridden by the roomstat?  Wouldn't you expect it to do exactly that?  If you have a single heating circuit (i.e. 1 zone) then you can't reliably have two devices whether physical or virtual to control it.  Maybe worth starting a thread yourself to see if folks here can help you with specifics?

(btw I **have** spent a fortune on Vaillant controls and it can be a nightmare not helped by Vaillant's very confusing product family names and product codes working out what is compatible but if you start off knowing what you want to achieve the Vaillant controls, while still clunky, are amongst the best you'll find and then ebusd adds extra icing!)

**Reply by sebska** [reply]
Probably not write it to the boiler but write it to a changeable value in the thermostat / steering device? Or: How would you change it without ebusd? Then you must try to change it there with ebusd...
